autodistill/cli.py

```diff
@@ -0,0 +1,152 @@
+@click.option("--epochs", default=200, required=True)
+@click.option("--output", default="./dataset", required=True)
+@click.option("--upload-to-roboflow", default=False, required=False)
+@click.option("--project_license", default="MIT", required=False)
+def main(
+    images, base, target, model_type, ontology, epochs, output, upload_to_roboflow, project_license
+):
+    if model_type not in SUPPORTED_MODEL_TYPES:
+        print(
+            "Model type not supported. Please choose from the following model types: "
+            + str(SUPPORTED_MODEL_TYPES)
+        )
+        exit()
+
+    if ontology == "{}":
+        print("No ontology provided. Please provide an ontology.")
+        exit()
+
+    print("Loading base model...")
+    model = import_requisite_module(base)
+
+    try:
+        ontology = json.loads(ontology, strict=False)
+    except:
+        print("Ontologies must be valid JSON.")
+        exit()
+
+    ontology = CaptionOntology(ontology)
+
+    base_model = model(ontology=ontology)
+
+    print("Labeling data...")
+
+    # if dir is file, run label on file
+    if os.path.isfile(images):
+        classes = ontology.classes()
+
+        box_annotator = sv.BoxAnnotator()
+
+        image = cv2.imread(images)
+
+        base_model.label(
+            input_folder=os.path.dirname(images),
+            extension=os.path.splitext(images)[1],
+            output_folder=output,
+        )
+        # show results
+        detections = base_model.predict(images)
+
+        labels = [
+            f"{classes[class_id]} {confidence:0.2f}"
+            for _, _, confidence, class_id, _ in detections
+        ]
+
+        annotated_frame = box_annotator.annotate(
+            scene=image.copy(),
+            detections=detections,
+            labels=labels,
+        )
+
+        sv.plot_image(annotated_frame, size=(8, 8))
+
+        exit()
+
+    base_model.label(
+        input_folder=dir,
+        output_folder=output
+    )
+
+    print("Loading target model...")
+    target_model = import_requisite_module(target)
+
+    print("Training target model...")
+    target_model.train(dataset_yaml=os.path.join(output, "data.yaml"), epochs=epochs)
+
+    if upload_to_roboflow:
+        roboflow.login()
+
+        rf = roboflow.Roboflow()
+
+        workspace = rf.workspace()
+
+        if model_type == "detection":
+            rf_model_value = "object-detection"
+        elif model_type == "segmentation":
+            rf_model_value = "image-segmentation"
+        else:
+            rf_model_value = "single-label-classification"
+
+        project = workspace.create_project(
+            "autodistill",
+            project_license=project_license,
+            annotation=ontology.classes()[0],
+            project_type=rf_model_value
+        )
+
+        rf.workspace().upload_dataset(
+            dataset_path=output,
+            project_name=project.id.split("/")[-1],
+            dataset_format="yolov8",
```

Comment:
See https://github.com/roboflow/roboflow-python/blob/5ae3e32c2225effb273eba89e1bf205ac1e2c480/roboflow/core/workspace.py#L251 for the function that's going on in the `roboflow` pip package.

You need to provide a YOLOv8, YOLOv5, or VOC dataset. Should we make that more explicit / add a dataset type inference method?

Reply:
Hmmmm, why is that a requirement? We support all sorts of formats 🤔 

But yes, we should give them a hint at least.

Reply:
Also the default shouldn't be yolov8; it's not a great format.

Reply:
Should we make VOC the default?

Reply:
Does our VOC format support polygons? COCO might be better (though I think we have issues because then it uploads the full dataset for each image)

No format is great I guess 😂

Reply:
I don't think we have COCO. I'm going to move this to Slack for discussion, since it affects `roboflow-python` and maybe the API.